react/features/conference/middleware.js

```diff
@@ -48,3 +51,25 @@ MiddlewareRegistry.register(store => next => action => {
 
     return result;
 });
+
+/**
+ * Set up state change listener to perform maintenance tasks when the conference
+ * is left or failed, close all dialogs and unpin any pinned participants.
+ */
+StateListenerRegistry.register(
+    state => getCurrentConference(state),
+    (conference, { dispatch, getState }, prevConference) => {
+        const { authRequired, passwordRequired }
+            = getState()['features/base/conference'];
+
+        if (conference !== prevConference) {
+            dispatch(pinParticipant(null));
```

Comment:
I tried to find where we clear the participant list when a meeting is left, but I couldn't find it with a 30sec search. Depending where that is, do you think it would be wise to do this kind of cleanup there? As the pinned participant info doesn't make sense, when there are no more participants in the conference.

Also, maybe a similar cleanup would be nice in the reducer that handles PARTICIPANT_LEFT, when the leaving participant is pinned?

Reply:
That is already taken care of somewhere. This adresses a single corner-case: pinning of the local participant. Otherwise becomes sticky, because the local participant object is reused.

Reply:
In this case can we add a clause here so then we don't do an unnecessary redux update, when it's not the local participant? or at least a comment why this is needed here

Reply:
Done.